### ISSUE TITLE
Wezterm not starting with wayland_enable=1 on Hyprland-git

### ISSUE THREAD

**Issue by ShadowS0ng** [wez/wezterm#5103]
### What Operating System(s) are you seeing this problem on?

Linux Wayland

### Which Wayland compositor or X11 Window manager(s) are you using?

Hyprland-git
v0.36.0-31-g328ab431

### WezTerm version

20240226-174525-22424c32

### Did you try the latest nightly build to see if the issue is better (or worse!) than your current version?

Yes, and I updated the version box above to show the version of the nightly that I tried

### Describe the bug

wezterm doesn't launch on Hyprland-git version after a commit on hyprland which i think does a size hint
WAYLAND_DEBUG=1 output:
```
[2072246.335]  -> wl_display@1.get_registry(new id wl_registry@2)
[2072246.369]  -> wl_display@1.sync(new id wl_callback@3)
[2072246.509] wl_display@1.delete_id(3)
[2072246.525] wl_registry@2.global(1, "wl_shm", 1)
[2072246.555] wl_registry@2.global(2, "wl_drm", 2)
[2072246.564] wl_registry@2.global(3, "zwp_linux_dmabuf_v1", 4)
[2072246.574] wl_registry@2.global(4, "wl_compositor", 6)
[2072246.583] wl_registry@2.global(5, "wl_subcompositor", 1)
[2072246.596] wl_registry@2.global(6, "wl_data_device_manager", 3)
[2072246.605] wl_registry@2.global(7, "zwlr_export_dmabuf_manager_v1", 1)
[2072246.616] wl_registry@2.global(8, "zwlr_data_control_manager_v1", 2)
[2072246.625] wl_registry@2.global(9, "zwp_primary_selection_device_manager_v1", 1)
[2072246.636] wl_registry@2.global(10, "wp_viewporter", 1)
[2072246.645] wl_registry@2.global(11, "zwlr_gamma_control_manager_v1", 1)
[2072246.654] wl_registry@2.global(12, "zwlr_output_power_manager_v1", 1)
[2072246.664] wl_registry@2.global(13, "xdg_wm_base", 6)
[2072246.674] wl_registry@2.global(14, "wl_seat", 9)
[2072246.683] wl_registry@2.global(15, "wp_presentation", 1)
[2072246.693] wl_registry@2.global(16, "ext_idle_notifier_v1", 1)
[2072246.702] wl_registry@2.global(17, "zwlr_layer_shell_v1", 4)
[2072246.712] wl_registry@2.global(18, "org_kde_kwin_server_decoration_manager", 1)
[2072246.722] wl_registry@2.global(19, "zxdg_decoration_manager_v1", 1)
[2072246.731] wl_registry@2.global(20, "zwlr_output_manager_v1", 4)
[2072246.740] wl_registry@2.global(21, "zwp_keyboard_shortcuts_inhibit_manager_v1", 1)
[2072246.750] wl_registry@2.global(22, "zwp_pointer_constraints_v1", 1)
[2072246.759] wl_registry@2.global(23, "zwp_relative_pointer_manager_v1", 1)
[2072246.769] wl_registry@2.global(24, "zwp_virtual_keyboard_manager_v1", 1)
[2072246.778] wl_registry@2.global(25, "zwlr_virtual_pointer_manager_v1", 2)
[2072246.787] wl_registry@2.global(26, "zwlr_foreign_toplevel_manager_v1", 3)
[2072246.797] wl_registry@2.global(27, "wp_drm_lease_device_v1", 1)
[2072246.806] wl_registry@2.global(28, "zwp_tablet_manager_v2", 1)
[2072246.816] wl_registry@2.global(29, "zwp_idle_inhibit_manager_v1", 1)
[2072246.825] wl_registry@2.global(30, "zxdg_exporter_v1", 1)
[2072246.835] wl_registry@2.global(31, "zxdg_importer_v1", 1)
[2072246.845] wl_registry@2.global(32, "zxdg_exporter_v2", 1)
[2072246.854] wl_registry@2.global(33, "zxdg_importer_v2", 1)
[2072246.864] wl_registry@2.global(34, "zwp_pointer_gestures_v1", 3)
[2072246.873] wl_registry@2.global(35, "zwp_text_input_manager_v3", 1)
[2072246.883] wl_registry@2.global(36, "zwp_input_method_manager_v2", 1)
[2072246.892] wl_registry@2.global(37, "xdg_activation_v1", 1)
[2072246.901] wl_registry@2.global(38, "ext_session_lock_manager_v1", 1)
[2072246.910] wl_registry@2.global(39, "wp_cursor_shape_manager_v1", 1)
[2072246.919] wl_registry@2.global(40, "wp_tearing_control_manager_v1", 1)
[2072246.927] wl_registry@2.global(41, "wp_single_pixel_buffer_manager_v1", 1)
[2072246.937] wl_registry@2.global(42, "xwayland_shell_v1", 1)
[2072246.946] wl_registry@2.global(43, "hyprland_toplevel_export_manager_v1", 2)
[2072246.954] wl_registry@2.global(44, "wp_fractional_scale_manager_v1", 1)
[2072246.963] wl_registry@2.global(45, "zwp_text_input_manager_v1", 1)
[2072246.972] wl_registry@2.global(46, "hyprland_global_shortcuts_manager_v1", 1)
[2072246.981] wl_registry@2.global(47, "zwlr_screencopy_manager_v1", 3)
[2072246.990] wl_registry@2.global(48, "zxdg_output_manager_v1", 3)
[2072246.999] wl_registry@2.global(49, "wl_output", 4)
[2072247.008] wl_callback@3.done(207)
[2072247.030]  -> wl_registry@2.bind(1, "wl_shm", 1, new id [unknown]@3)
[2072247.070]  -> wl_shm@3.create_pool(new id wl_shm_pool@4, fd 11, 1)
[2072247.100]  -> wl_registry@2.bind(4, "wl_compositor", 5, new id [unknown]@5)
[2072247.113]  -> wl_registry@2.bind(5, "wl_subcompositor", 1, new id [unknown]@6)
[2072247.131]  -> wl_registry@2.bind(49, "wl_output", 4, new id [unknown]@7)
[2072247.143]  -> wl_registry@2.bind(48, "zxdg_output_manager_v1", 3, new id [unknown]@8)
[2072247.161]  -> zxdg_output_manager_v1@8.get_xdg_output(new id zxdg_output_v1@9, wl_output@7)
[2072247.179]  -> wl_registry@2.bind(35, "zwp_text_input_manager_v3", 1, new id [unknown]@10)
[2072247.191]  -> wl_registry@2.bind(14, "wl_seat", 7, new id [unknown]@11)
[2072247.203]  -> wl_registry@2.bind(13, "xdg_wm_base", 5, new id [unknown]@12)
[2072247.214]  -> wl_registry@2.bind(19, "zxdg_decoration_manager_v1", 1, new id [unknown]@13)
[2072247.225]  -> wl_registry@2.bind(6, "wl_data_device_manager", 3, new id [unknown]@14)
[2072247.236]  -> wl_registry@2.bind(9, "zwp_primary_selection_device_manager_v1", 1, new id [unknown]@15)
[2072390.996] wl_shm@3.format(0)
[2072391.020] wl_shm@3.format(1)
[2072391.032] wl_shm@3.format(875709016)
[2072391.040] wl_shm@3.format(875708993)
[2072391.061] wl_shm@3.format(875710274)
[2072391.071] wl_shm@3.format(842094674)
[2072391.078] wl_shm@3.format(842088786)
[2072391.084] wl_shm@3.format(892426322)
[2072391.089] wl_shm@3.format(892420434)
[2072391.095] wl_shm@3.format(909199186)
[2072391.100] wl_shm@3.format(808665688)
[2072391.106] wl_shm@3.format(808665665)
[2072391.112] wl_shm@3.format(1211384408)
[2072391.123] wl_shm@3.format(1211384385)
[2072391.133] wl_output@7.geometry(0, 0, 340, 190, 2, "AU Optronics", "0x45EC", 0)
[2072391.150] wl_output@7.mode(1, 1366, 768, 60001)
[2072391.161] wl_output@7.scale(1)
[2072391.179] wl_output@7.name("LVDS-1")
[2072391.189] wl_output@7.description("AU Optronics 0x45EC 0x00000001 (LVDS-1)")
[2072391.197] wl_output@7.done()
[2072391.204] zxdg_output_v1@9.name("LVDS-1")
[2072391.213] zxdg_output_v1@9.description("AU Optronics 0x45EC 0x00000001 (LVDS-1)")
[2072391.221] zxdg_output_v1@9.logical_position(0, 0)
[2072391.229] zxdg_output_v1@9.logical_size(1366, 768)
[2072391.237] wl_output@7.done()
[2072391.243] wl_seat@11.name("seat0")
[2072391.251] wl_seat@11.capabilities(3)
[2072391.482]  -> wl_seat@11.get_keyboard(new id wl_keyboard@16)
[2072391.510]  -> zwp_text_input_manager_v3@10.get_text_input(new id zwp_text_input_v3@17, wl_seat@11)
[2072391.526]  -> wl_seat@11.get_pointer(new id wl_pointer@18)
[2072391.541]  -> wl_data_device_manager@14.get_data_device(new id wl_data_device@19, wl_seat@11)
[2072391.551]  -> zwp_primary_selection_device_manager_v1@15.get_device(new id zwp_primary_selection_device_v1@20, wl_seat@11)
[2072430.919]  -> wl_compositor@5.create_surface(new id wl_surface@21)
[2072430.945]  -> wl_compositor@5.create_surface(new id wl_surface@22)
[2072430.967]  -> xdg_wm_base@12.get_xdg_surface(new id xdg_surface@23, wl_surface@21)
[2072430.980]  -> xdg_surface@23.get_toplevel(new id xdg_toplevel@24)
[2072430.992]  -> zxdg_decoration_manager_v1@13.get_toplevel_decoration(new id zxdg_toplevel_decoration_v1@25, xdg_toplevel@24)
[2072431.003]  -> zxdg_toplevel_decoration_v1@25.set_mode(2)
[2072431.016]  -> xdg_toplevel@24.set_app_id("org.wezfurlong.wezterm")
[2072431.026]  -> xdg_toplevel@24.set_title("wezterm")
[2072431.040]  -> zxdg_toplevel_decoration_v1@25.set_mode(2)
[2072431.087]  -> wl_shm@3.create_pool(new id wl_shm_pool@26, fd 23, 1)
[2072431.123]  -> wl_compositor@5.create_surface(new id wl_surface@27)
[2072431.139]  -> wl_subcompositor@6.get_subsurface(new id wl_subsurface@28, wl_surface@27, wl_surface@21)
[2072431.152]  -> wl_subsurface@28.set_sync()
[2072431.163]  -> wl_compositor@5.create_surface(new id wl_surface@29)
[2072431.175]  -> wl_subcompositor@6.get_subsurface(new id wl_subsurface@30, wl_surface@29, wl_surface@21)
[2072431.192]  -> wl_subsurface@30.set_sync()
[2072431.202]  -> wl_compositor@5.create_surface(new id wl_surface@31)
[2072431.215]  -> wl_subcompositor@6.get_subsurface(new id wl_subsurface@32, wl_surface@31, wl_surface@21)
[2072431.226]  -> wl_subsurface@32.set_sync()
[2072431.237]  -> wl_compositor@5.create_surface(new id wl_surface@33)
[2072431.248]  -> wl_subcompositor@6.get_subsurface(new id wl_subsurface@34, wl_surface@33, wl_surface@21)
[2072431.260]  -> wl_subsurface@34.set_sync()
[2072431.269]  -> wl_compositor@5.create_surface(new id wl_surface@35)
[2072431.280]  -> wl_subcompositor@6.get_subsurface(new id wl_subsurface@36, wl_surface@35, wl_surface@21)
[2072431.291]  -> wl_subsurface@36.set_sync()
[2072431.302]  -> wl_subsurface@28.destroy()
[2072431.313]  -> wl_surface@27.destroy()
[2072431.331]  -> wl_subsurface@30.destroy()
[2072431.341]  -> wl_surface@29.destroy()
[2072431.351]  -> wl_subsurface@32.destroy()
[2072431.360]  -> wl_surface@31.destroy()
[2072431.369]  -> wl_subsurface@34.destroy()
[2072431.378]  -> wl_surface@33.destroy()
[2072431.388]  -> wl_subsurface@36.destroy()
[2072431.398]  -> wl_surface@35.destroy()
[2072431.409]  -> xdg_toplevel@24.set_min_size(32, 32)
[2072431.421]  -> xdg_surface@23.set_window_geometry(0, 0, 945, 629)
[2072431.433]  -> wl_surface@21.commit()
[2072431.501] wl_keyboard@16.keymap(1, fd 23, 70269)
[2072431.515] wl_keyboard@16.repeat_info(50, 600)
[2072447.354] wl_display@1.delete_id(28)
[2072447.375] wl_display@1.delete_id(27)
[2072447.382] wl_display@1.delete_id(30)
[2072447.388] wl_display@1.delete_id(29)
[2072447.399] wl_display@1.delete_id(32)
[2072447.405] wl_display@1.delete_id(31)
[2072447.409] wl_display@1.delete_id(34)
[2072447.412] wl_display@1.delete_id(33)
[2072447.416] wl_display@1.delete_id(36)
[2072447.419] wl_display@1.delete_id(35)
[2072447.423] xdg_toplevel@24.wm_capabilities(array[16])
[2072447.429] xdg_toplevel@24.configure(1366, 730, array[0])
[2072447.436] zxdg_toplevel_decoration_v1@25.configure(2)
[2072447.447] xdg_surface@23.configure(208)
[2072447.481]  -> xdg_surface@23.ack_configure(208)
[2072447.583]  -> xdg_surface@23.set_window_geometry(0, 0, 1366, 730)
[2072486.653]  -> wl_surface@21.commit()
[2072486.678]  -> wl_surface@21.frame(new id wl_callback@35)
[2072495.170]  -> wl_display@1.get_registry(new id wl_registry@36)
[2072495.190]  -> wl_display@1.sync(new id wl_callback@33)
[2072495.359] wl_display@1.delete_id(33)
[2072495.374] wl_registry@36.global(1, "wl_shm", 1)
[2072495.387] wl_registry@36.global(2, "wl_drm", 2)
[2072495.398]  -> wl_registry@36.bind(2, "wl_drm", 2, new id [unknown]@34)
[2072495.408] wl_registry@36.global(3, "zwp_linux_dmabuf_v1", 4)
[2072495.418] wl_registry@36.global(4, "wl_compositor", 6)
[2072495.427] wl_registry@36.global(5, "wl_subcompositor", 1)
[2072495.437] wl_registry@36.global(6, "wl_data_device_manager", 3)
[2072495.446] wl_registry@36.global(7, "zwlr_export_dmabuf_manager_v1", 1)
[2072495.456] wl_registry@36.global(8, "zwlr_data_control_manager_v1", 2)
[2072495.466] wl_registry@36.global(9, "zwp_primary_selection_device_manager_v1", 1)
[2072495.475] wl_registry@36.global(10, "wp_viewporter", 1)
[2072495.485] wl_registry@36.global(11, "zwlr_gamma_control_manager_v1", 1)
[2072495.494] wl_registry@36.global(12, "zwlr_output_power_manager_v1", 1)
[2072495.504] wl_registry@36.global(13, "xdg_wm_base", 6)
[2072495.513] wl_registry@36.global(14, "wl_seat", 9)
[2072495.522] wl_registry@36.global(15, "wp_presentation", 1)
[2072495.532] wl_registry@36.global(16, "ext_idle_notifier_v1", 1)
[2072495.541] wl_registry@36.global(17, "zwlr_layer_shell_v1", 4)
[2072495.549] wl_registry@36.global(18, "org_kde_kwin_server_decoration_manager", 1)
[2072495.559] wl_registry@36.global(19, "zxdg_decoration_manager_v1", 1)
[2072495.568] wl_registry@36.global(20, "zwlr_output_manager_v1", 4)
[2072495.577] wl_registry@36.global(21, "zwp_keyboard_shortcuts_inhibit_manager_v1", 1)
[2072495.587] wl_registry@36.global(22, "zwp_pointer_constraints_v1", 1)
[2072495.597] wl_registry@36.global(23, "zwp_relative_pointer_manager_v1", 1)
[2072495.606] wl_registry@36.global(24, "zwp_virtual_keyboard_manager_v1", 1)
[2072495.616] wl_registry@36.global(25, "zwlr_virtual_pointer_manager_v1", 2)
[2072495.625] wl_registry@36.global(26, "zwlr_foreign_toplevel_manager_v1", 3)
[2072495.634] wl_registry@36.global(27, "wp_drm_lease_device_v1", 1)
[2072495.643] wl_registry@36.global(28, "zwp_tablet_manager_v2", 1)
[2072495.652] wl_registry@36.global(29, "zwp_idle_inhibit_manager_v1", 1)
[2072495.661] wl_registry@36.global(30, "zxdg_exporter_v1", 1)
[2072495.668] wl_registry@36.global(31, "zxdg_importer_v1", 1)
[2072495.675] wl_registry@36.global(32, "zxdg_exporter_v2", 1)
[2072495.681] wl_registry@36.global(33, "zxdg_importer_v2", 1)
[2072495.688] wl_registry@36.global(34, "zwp_pointer_gestures_v1", 3)
[2072495.694] wl_registry@36.global(35, "zwp_text_input_manager_v3", 1)
[2072495.701] wl_registry@36.global(36, "zwp_input_method_manager_v2", 1)
[2072495.707] wl_registry@36.global(37, "xdg_activation_v1", 1)
[2072495.714] wl_registry@36.global(38, "ext_session_lock_manager_v1", 1)
[2072495.721] wl_registry@36.global(39, "wp_cursor_shape_manager_v1", 1)
[2072495.728] wl_registry@36.global(40, "wp_tearing_control_manager_v1", 1)
[2072495.735] wl_registry@36.global(41, "wp_single_pixel_buffer_manager_v1", 1)
[2072495.747] wl_registry@36.global(42, "xwayland_shell_v1", 1)
[2072495.756] wl_registry@36.global(43, "hyprland_toplevel_export_manager_v1", 2)
[2072495.766] wl_registry@36.global(44, "wp_fractional_scale_manager_v1", 1)
[2072495.775] wl_registry@36.global(45, "zwp_text_input_manager_v1", 1)
[2072495.784] wl_registry@36.global(46, "hyprland_global_shortcuts_manager_v1", 1)
[2072495.794] wl_registry@36.global(47, "zwlr_screencopy_manager_v1", 3)
[2072495.803] wl_registry@36.global(48, "zxdg_output_manager_v1", 3)
[2072495.812] wl_registry@36.global(49, "wl_output", 4)
[2072495.822] wl_callback@33.done(208)
[2072495.833]  -> wl_display@1.sync(new id wl_callback@33)
[2072495.955] wl_display@1.delete_id(33)
[2072495.965] wl_drm@34.device("/dev/dri/renderD128")
[2072495.975] wl_drm@34.capabilities(1)
[2072495.984] wl_drm@34.format(1211384385)
[2072495.993] wl_drm@34.format(1211384408)
[2072496.002] wl_drm@34.format(942948929)
[2072496.010] wl_drm@34.format(942948952)
[2072496.018] wl_drm@34.format(808669761)
[2072496.026] wl_drm@34.format(808669784)
[2072496.034] wl_drm@34.format(808665665)
[2072496.042] wl_drm@34.format(875713089)
[2072496.049] wl_drm@34.format(875708993)
[2072496.055] wl_drm@34.format(875713112)
[2072496.064] wl_drm@34.format(875709016)
[2072496.072] wl_drm@34.format(892424769)
[2072496.080] wl_drm@34.format(842093121)
[2072496.088] wl_drm@34.format(909199186)
[2072496.096] wl_drm@34.format(538982482)
[2072496.104] wl_drm@34.format(540422482)
[2072496.112] wl_drm@34.format(943215175)
[2072496.120] wl_drm@34.format(842224199)
[2072496.128] wl_drm@34.format(961959257)
[2072496.136] wl_drm@34.format(825316697)
[2072496.145] wl_drm@34.format(842093913)
[2072496.152] wl_drm@34.format(909202777)
[2072496.159] wl_drm@34.format(875713881)
[2072496.166] wl_drm@34.format(961893977)
[2072496.173] wl_drm@34.format(825316953)
[2072496.183] wl_drm@34.format(842094169)
[2072496.190] wl_drm@34.format(909203033)
[2072496.198] wl_drm@34.format(875714137)
[2072496.205] wl_drm@34.format(842094158)
[2072496.212] wl_drm@34.format(825382478)
[2072496.218] wl_drm@34.format(808530000)
[2072496.225] wl_drm@34.format(842084432)
[2072496.231] wl_drm@34.format(909193296)
[2072496.238] wl_drm@34.format(808661072)
[2072496.244] wl_drm@34.format(909203022)
[2072496.251] wl_drm@34.format(1448433985)
[2072496.257] wl_drm@34.format(1448434008)
[2072496.264] wl_drm@34.format(808531033)
[2072496.270] wl_drm@34.format(842085465)
[2072496.276] wl_drm@34.format(909194329)
[2072496.282] wl_drm@34.format(1448695129)
[2072496.289] wl_drm@34.format(1431918169)
[2072496.296] wl_drm@34.format(1498831189)
[2072496.302] wl_drm@34.format(1498765654)
[2072496.308] wl_drm@34.format(808530521)
[2072496.314] wl_drm@34.format(842084953)
[2072496.321] wl_drm@34.format(909193817)
[2072496.327] wl_callback@33.done(208)
[2072497.953]  -> wl_display@1.get_registry(new id wl_registry@33)
[2072497.978]  -> wl_display@1.sync(new id wl_callback@31)
[2072498.122] wl_display@1.delete_id(31)
[2072498.144] wl_registry@33.global(1, "wl_shm", 1)
[2072498.153] wl_registry@33.global(2, "wl_drm", 2)
[2072498.160] wl_registry@33.global(3, "zwp_linux_dmabuf_v1", 4)
[2072498.169]  -> wl_registry@33.bind(3, "zwp_linux_dmabuf_v1", 4, new id [unknown]@32)
[2072498.178] wl_registry@33.global(4, "wl_compositor", 6)
[2072498.186] wl_registry@33.global(5, "wl_subcompositor", 1)
[2072498.194] wl_registry@33.global(6, "wl_data_device_manager", 3)
[2072498.202] wl_registry@33.global(7, "zwlr_export_dmabuf_manager_v1", 1)
[2072498.213] wl_registry@33.global(8, "zwlr_data_control_manager_v1", 2)
[2072498.227] wl_registry@33.global(9, "zwp_primary_selection_device_manager_v1", 1)
[2072498.235] wl_registry@33.global(10, "wp_viewporter", 1)
[2072498.242] wl_registry@33.global(11, "zwlr_gamma_control_manager_v1", 1)
[2072498.250] wl_registry@33.global(12, "zwlr_output_power_manager_v1", 1)
[2072498.257] wl_registry@33.global(13, "xdg_wm_base", 6)
[2072498.265] wl_registry@33.global(14, "wl_seat", 9)
[2072498.272] wl_registry@33.global(15, "wp_presentation", 1)
[2072498.279] wl_registry@33.global(16, "ext_idle_notifier_v1", 1)
[2072498.287] wl_registry@33.global(17, "zwlr_layer_shell_v1", 4)
[2072498.294] wl_registry@33.global(18, "org_kde_kwin_server_decoration_manager", 1)
[2072498.302] wl_registry@33.global(19, "zxdg_decoration_manager_v1", 1)
[2072498.309] wl_registry@33.global(20, "zwlr_output_manager_v1", 4)
[2072498.316] wl_registry@33.global(21, "zwp_keyboard_shortcuts_inhibit_manager_v1", 1)
[2072498.324] wl_registry@33.global(22, "zwp_pointer_constraints_v1", 1)
[2072498.331] wl_registry@33.global(23, "zwp_relative_pointer_manager_v1", 1)
[2072498.339] wl_registry@33.global(24, "zwp_virtual_keyboard_manager_v1", 1)
[2072498.346] wl_registry@33.global(25, "zwlr_virtual_pointer_manager_v1", 2)
[2072498.354] wl_registry@33.global(26, "zwlr_foreign_toplevel_manager_v1", 3)
[2072498.361] wl_registry@33.global(27, "wp_drm_lease_device_v1", 1)
[2072498.368] wl_registry@33.global(28, "zwp_tablet_manager_v2", 1)
[2072498.376] wl_registry@33.global(29, "zwp_idle_inhibit_manager_v1", 1)
[2072498.383] wl_registry@33.global(30, "zxdg_exporter_v1", 1)
[2072498.390] wl_registry@33.global(31, "zxdg_importer_v1", 1)
[2072498.398] wl_registry@33.global(32, "zxdg_exporter_v2", 1)
[2072498.405] wl_registry@33.global(33, "zxdg_importer_v2", 1)
[2072498.412] wl_registry@33.global(34, "zwp_pointer_gestures_v1", 3)
[2072498.420] wl_registry@33.global(35, "zwp_text_input_manager_v3", 1)
[2072498.428] wl_registry@33.global(36, "zwp_input_method_manager_v2", 1)
[2072498.436] wl_registry@33.global(37, "xdg_activation_v1", 1)
[2072498.443] wl_registry@33.global(38, "ext_session_lock_manager_v1", 1)
[2072498.451] wl_registry@33.global(39, "wp_cursor_shape_manager_v1", 1)
[2072498.459] wl_registry@33.global(40, "wp_tearing_control_manager_v1", 1)
[2072498.466] wl_registry@33.global(41, "wp_single_pixel_buffer_manager_v1", 1)
[2072498.474] wl_registry@33.global(42, "xwayland_shell_v1", 1)
[2072498.482] wl_registry@33.global(43, "hyprland_toplevel_export_manager_v1", 2)
[2072498.490] wl_registry@33.global(44, "wp_fractional_scale_manager_v1", 1)
[2072498.497] wl_registry@33.global(45, "zwp_text_input_manager_v1", 1)
[2072498.504] wl_registry@33.global(46, "hyprland_global_shortcuts_manager_v1", 1)
[2072498.512] wl_registry@33.global(47, "zwlr_screencopy_manager_v1", 3)
[2072498.520] wl_registry@33.global(48, "zxdg_output_manager_v1", 3)
[2072498.527] wl_registry@33.global(49, "wl_output", 4)
[2072498.535] wl_callback@31.done(208)
[2072498.544]  -> zwp_linux_dmabuf_v1@32.get_default_feedback(new id zwp_linux_dmabuf_feedback_v1@31)
[2072498.552]  -> wl_display@1.sync(new id wl_callback@29)
[2072498.633] wl_display@1.delete_id(29)
[2072498.643] zwp_linux_dmabuf_feedback_v1@31.main_device(array[8])
[2072499.416] zwp_linux_dmabuf_feedback_v1@31.format_table(fd 23, 3008)
[2072499.445] zwp_linux_dmabuf_feedback_v1@31.tranche_target_device(array[8])
[2072499.454] zwp_linux_dmabuf_feedback_v1@31.tranche_flags(0)
[2072499.464] zwp_linux_dmabuf_feedback_v1@31.tranche_formats(array[376])
[2072499.486] zwp_linux_dmabuf_feedback_v1@31.tranche_done()
[2072499.493] zwp_linux_dmabuf_feedback_v1@31.done()
[2072499.502] wl_callback@29.done(208)
[2072499.511]  -> zwp_linux_dmabuf_feedback_v1@31.destroy()
[2072549.408]  -> zwp_linux_dmabuf_v1@32.get_surface_feedback(new id zwp_linux_dmabuf_feedback_v1@29, wl_surface@21)
[2072549.441]  -> wl_display@1.sync(new id wl_callback@30)
[2072549.565] wl_display@1.delete_id(31)
[2072549.580] wl_display@1.delete_id(30)
[2072549.591] wl_callback@30.done(208)
[2072552.261] zwp_linux_dmabuf_feedback_v1@29.main_device(array[8])
[2072552.290] zwp_linux_dmabuf_feedback_v1@29.format_table(fd 27, 3008)
[2072552.312] zwp_linux_dmabuf_feedback_v1@29.tranche_target_device(array[8])
[2072552.324] zwp_linux_dmabuf_feedback_v1@29.tranche_flags(0)
[2072552.331] zwp_linux_dmabuf_feedback_v1@29.tranche_formats(array[376])
[2072552.342] zwp_linux_dmabuf_feedback_v1@29.tranche_done()
[2072552.355] zwp_linux_dmabuf_feedback_v1@29.done()
[2072916.923]  -> xdg_toplevel@24.set_title(" ~")
[2072916.955]  -> wl_surface@21.commit()
```

### To Reproduce

install hyprland git version and run wezterm

### Configuration

config.warn_about_missing_glyphs = true
config.enable_wayland = true
config.audible_bell = "SystemBeep"
-- config.prefer_egl = true
config.enable_scroll_bar = true
config.initial_cols = 103
config.front_end = "OpenGL"
config.initial_rows = 29
-- config.window_background_opacity = 0.8
config.line_height = 1.1
-- config.cell_width = 1.1

config.hide_tab_bar_if_only_one_tab = true
config.xcursor_theme = "Bibata-Modern-Classic"
config.xcursor_size = 22
config.default_cursor_style = "BlinkingBlock"
config.cursor_blink_rate = 800

**Comment by nonetrix** [comment]
Same

**Comment by ShadowS0ng** [comment]
> Same, is this a issue with Hyprland or Wezterm though?

I opened an issue on Hyprland's github , it turns out apparently the problem is on wezterm's side .

**Comment by nonetrix** [comment]
So which version of Hyprland should I downgrade to? Sorry

**Comment by juvevood** [comment]
maybe need wezterm support the new wlroots protocol

**Comment by ShadowS0ng** [comment]
> So which version of Hyprland should I downgrade to? Sorry

on release version 36.0 it works fine

**Comment by ErrorNoInternet** [comment]
The Hyprland commit that "broke" WezTerm was hyprwm/Hyprland@f4f3aa2e505ca1b22688fb4a32f31c4bfb851dd0. All it does is send an initial size to the application so that it wouldn't need to resize itself and cause visual artifacts on startup.

WezTerm still runs if you open another terminal (e.g. kitty), make it float, then run `wezterm` inside that floating terminal.

**Comment by ShadowS0ng** [comment]
> The Hyprland commit that "broke" WezTerm was [hyprwm/Hyprland@f4f3aa2](https://github.com/hyprwm/Hyprland/commit/f4f3aa2e505ca1b22688fb4a32f31c4bfb851dd0). All it does is send an initial size to the application so that it wouldn't need to resize itself again later and cause visual artifacts on startup.
> 
> WezTerm still runs if you open another terminal (e.g. kitty), make it float, then launch `wezterm` from that terminal. Doesn't work if you run it from an app launcher for some reason.

in my case it doesn't start from a floating terminal but if when i added a rule to hyprland's config to float wezterm
(i.e : `windowrulev2 = float,class:^(org.wezfurlong.wezterm)$`), it starts in floating mode without problem.

**Comment by RougeEtoile** [comment]
>WezTerm still runs for me if I open another terminal (e.g. kitty), make it float, then launch wezterm from that terminal. Doesn't work if I run it from an app launcher for some reason.
This works for me

> in my case it doesn't start from a floating terminal but when i added a rule to hyprland's config to float wezterm (i.e : `windowrulev2 = float,class:^(org.wezfurlong.wezterm)$`), it starts in floating mode without problem.
This does not work for me

**Comment by RougeEtoile** [comment]
https://github.com/hyprwm/Hyprland/issues/4806

**Comment by deviantsemicolon** [comment]
wezterm just doesn't work for me at all on hyprland even if I start it from another terminal. It just doesn't open

**Comment by LeoGitHuber** [comment]
My solution is to build Wezterm without Wayland support.

**Comment by ErrorNoInternet** [comment]
Mine was to patch Hyprland to get rid of that commit: https://github.com/ErrorNoInternet/configuration.nix/blob/main/packages/hyprland/remove-size-predictions.patch

**Comment by nonetrix** [comment]
Hm might look into that, I wonder how I could apply it to the NixOS flake though

**Comment by ShadowS0ng** [comment]
> My solution is to build Wezterm without Wayland support.

you can explicitly declare disabling wayland support in config though. 
`config.enable_wayland = false`

**Comment by LeoGitHuber** [comment]
> > My solution is to build Wezterm without Wayland support.
> 
> you can explicitly declare disabling wayland support in config though. `config.enable_wayland = false`

Yep. I just forgot this setting. But thanks for your suggestion.

**Comment by nonetrix** [comment]
> Hm might look into that, I wonder how I could apply it to the NixOS flake though

I think there was some confusion, I was referring to the revert patch

**Comment by topaxi** [comment]
Hyprland 0.37 just hit arch extra, I assume this will hit a lot more users now.

Thank you @ShadowS0ng for the workaround :partying_face:

**Comment by yutkat** [comment]
I have worked around this problem by using Ulauncher. https://github.com/Ulauncher/Ulauncher

```
bind=SUPER,RETURN,exec,ulauncher-toggle -q "wezterm$"
```

I don't know the cause, but if I set `enable_wayland = false` in my environment, I'm getting a slow mouse cursor movement behavior.

**Comment by JeffDess** [comment]
I've managed to make it work with Wayland enabled with `WAYLAND_DISPLAY=1 wezterm`.

**Comment by ErrorNoInternet** [comment]
> I've managed to make it work with Wayland enabled with `WAYLAND_DISPLAY=1 wezterm`.

That just makes wezterm run under XWayland, because there is no Wayland display called "1", it should at least be "wayland-1". `WAYLAND_DISPLAY=hellothisisanonexistentdisplay` would also work.

**Comment by nonetrix** [comment]
So when will their be a PR to fix this?

**Comment by ShadowS0ng** [comment]
## UPDATE:
as a workaround:
so i added another rule to the hyprland config to tile wezterm after floating it , now it works fine in tiling mode :
`
windowrulev2 = float,class:^(org.wezfurlong.wezterm)$
windowrulev2 = tile,class:^(org.wezfurlong.wezterm)$
`

**Comment by juvevood** [comment]
I have switched to the foot terminal. its works fine with wayland.

**Comment by nonetrix** [comment]
I am having issue now that Wezterm on running on XWayland randomly becomes super slow for no reason, also seems to bug out Hyprland sometimes and make the Hyprland process use 100% of one core somehow making the mouse cursor lag

**Comment by nonetrix** [comment]
> ## UPDATE:
> 
> as a workaround: so i added another rule to the hyprland config to tile wezterm after floating it , now it works fine in tiling mode :
> 
> ```
> windowrulev2 = float,class:^(org.wezfurlong.wezterm)$
> windowrulev2 = tile,class:^(org.wezfurlong.wezterm)$
> ```

This does nothing for me

**Comment by fioncat** [comment]
Same, wezterm goes really slow in XWayland, especially when using neovim.

Now I had to downgrade hyprland to `0.35.0`, hope maintainer can fix this soon.

**Comment by nonetrix** [comment]
I imagine it being slow has to be Hyprland issue with XWayland though rather than Wezterm

**Comment by ErrorNoInternet** [comment]
Wrote an explanation here: https://github.com/wez/wezterm/issues/5197#issuecomment-2020263016

**Comment by JoshElias** [comment]
For the Hyprland users we currently have 4 workarounds for this issue before Wezterm gets updated. Which will probably be soon since @ErrorNoInternet has identified the issue.

- Revert to Hyprland 0.36
- Disable Wayland on Wezterm
    - `enable_wayland = false` in `wezterm.lua`
    - launch with `WAYLAND_DISPLAY=1 wezterm`
- Start Weztem in a floating window
- Use another terminal

**Comment by ShadowS0ng** [comment]
> > ## UPDATE:
> > as a workaround: so i added another rule to the hyprland config to tile wezterm after floating it , now it works fine in tiling mode :
> > ```
> > windowrulev2 = float,class:^(org.wezfurlong.wezterm)$
> > windowrulev2 = tile,class:^(org.wezfurlong.wezterm)$
> > ```
> 
> This does nothing for me

I suppose a proper fix will be out soon since this update hit the arch repos 1 week ago or so.

**Comment by dylantompkins** [comment]
this is a duplicate of #5067

**Comment by MrGamerDoesGames** [comment]
> ## UPDATE:
> as a workaround: so i added another rule to the hyprland config to tile wezterm after floating it , now it works fine in tiling mode :
> 
> ```
> windowrulev2 = float,class:^(org.wezfurlong.wezterm)$
> windowrulev2 = tile,class:^(org.wezfurlong.wezterm)$
> ```

Legend. This worked for me, and even though it's a janky fix, by god it works. Godspeed

**Comment by constantitus** [comment]
So this issue is 100% on wezterm. For anyone waiting on hyprland for a fix, it won't happen.
https://github.com/hyprwm/Hyprland/issues/4901#issuecomment-1972106835
Starting wezterm in floating mode seems to work, but that's an extremely inconvenient thing to do on a tiling window manager, so it's out of the question.
Using wezterm with `enable_wayland = false` is also out of the question as wezterm is awfully laggy in xwayland.
I'm sick of these half-a$$ed solutions. I'll patch hyprland for this release and hopefully wezterm fixes their wayland implementation until the next hyprland release, as this is not the first time wezterm's been having wayland issues.

**Comment by rosshadden** [comment]
@constantitus No need to be rude. We knew it was in wezterm's court, and I don't think any of us in here that were keeping tabs on the matter were "waiting on hyprland for a fix".

Perhaps you are new to using and contributing to open source projects, so I can help shed some light on how things work: You are welcome to fix wezterm for all of us, as am I, as is Wez himself. The reason it hasn't been fixed yet is simply that none of us have done so yet. I don't know about you @constantitus but I'm not exactly swimming in free time. Are you? Please by all means fix this for us :pray: :heart:.

**Comment by nonetrix** [comment]
But it running badly in XWayland honestly could be Hyprland issue, not sure has anyone had similar experience on Sway for example?

**Comment by fioncat** [comment]
No need to remove the original logic, all we need to do is sending a `NeedRepaint` to Wayland when `frame_callback` is not none:

```rust
        if self.frame_callback.is_some() {
            // Painting now won't be productive, so skip it but
            // remember that we need to be painted so that when
            // the compositor is ready for us, we can paint then.
            self.events.dispatch(WindowEvent::NeedRepaint); // Fix this issue.
            self.invalidated = true;
            return Ok(());
        }
```

If this is correct, I can open a PR to fix this.

**Comment by JoshElias** [comment]
@constantitus nobody is forcing you to use Wezterm. If you're sick of it, use another terminal. There's no shortage of options. Wezterm has zero obligation to work for you or anybody else. Says right on their readme this project is done on people's spare time.

**Comment by fioncat** [comment]
Add `self.events.dispatch(WindowEvent::NeedRepaint)` in `show` can fix this. I will open a PR later if I have time.

**Comment by ShadowS0ng** [comment]
> Even if most don't run it in XWayland if Wayland works someone should investigate that too, maybe someone should make another issue for it. I think it might be Hyprland though since the Hyprland process starts to use a lot of CPU and my mouse cursor stutters could be issue with other applications we don't know about
hyprland has a bad xwayland implementation , I mentioned that to vaxry , it's improved alot but it's still laggy especially for wine applications .

**Comment by frabjous** [comment]
From what I can tell, using the kludge:

`config.enable_wayland = false`

means that OSC52 copy/paste does not work, which makes wezterm unusable at present given my workflow.

I hope this gets fixed, as I really enjoy using wezterm otherwise and am very grateful for all the hard work that has gone into it.

**Comment by khjorth** [comment]
> > ## UPDATE:
> > as a workaround: so i added another rule to the hyprland config to tile wezterm after floating it , now it works fine in tiling mode :
> > ```
> > windowrulev2 = float,class:^(org.wezfurlong.wezterm)$
> > windowrulev2 = tile,class:^(org.wezfurlong.wezterm)$
> > ```
> 
> This does nothing for me

Same here, but the following bind *does* work for me, so perhaps it's worth a shot:

```
"$mod, return, exec, [float;tile] wezterm start --always-new-process"
```

**Comment by flakusha** [comment]
I have added temporary patch in .wezterm.lua until it's resolved.
As long as I only have sway and hyprland on my machine, it worked out fine.

```lua
if os.getenv("XDG_CURRENT_DESKTOP") == "Hyprland" then
	config.enable_wayland = false
else
	config.enable_wayland = true
end
```

**Comment by paulcarroty** [comment]
Workaround for Sway 1.9: `wezterm --config='front_end = "WebGpu"'`

**Comment by rdvm** [comment]
> > > ## UPDATE:
> > > as a workaround: so i added another rule to the hyprland config to tile wezterm after floating it , now it works fine in tiling mode :
> > > ```
> > > windowrulev2 = float,class:^(org.wezfurlong.wezterm)$
> > > windowrulev2 = tile,class:^(org.wezfurlong.wezterm)$
> > > ```
> > 
> > 
> > This does nothing for me
> 
> Same here, but the following bind _does_ work for me, so perhaps it's worth a shot:
> 
> ```
> "$mod, return, exec, [float;tile] wezterm start --always-new-process"
> ```

This has been working flawlessly for me since I saw it a few weeks ago. Thanks for sharing, @khjorth !

**Comment by inudev3** [comment]
is this fixed? i am on nightly build and still errors.

**Comment by ErrorNoInternet** [comment]
> is this fixed? i am on nightly build and still errors.

If you're getting errors and/or wezterm immediately closes that's a different issue. This one (wezterm run normally but no window appears) has been fixed.

**Comment by raffaem** [comment]
@wez Would you consider a new release? This is affecting lots of users it seems.

**Comment by offeex** [comment]
can confirm. Doesn't start at all, even with floating window workaround. Using hyprland 0.41 on Arch

**Comment by JimitSoni18** [comment]
+1 does not start at all, check the attached picture, i have opened wezterm in the shell by typing wezterm, it does not output anything, nor does any window open up
![20240706_14h23m05s_grim](https://github.com/wez/wezterm/assets/76764841/9d5c2106-30cb-4aeb-89b2-999ef2926ae1)

**Comment by ShadowS0ng** [comment]
> +1 does not start at all, check the attached picture, i have opened wezterm in the shell by typing wezterm, it does not output anything, nor does any window open up (i have installed wezterm as a void package using xbps) ![20240706_14h23m05s_grim](https://private-user-images.githubusercontent.com/76764841/346263321-9d5c2106-30cb-4aeb-89b2-999ef2926ae1.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjAyODI3NjgsIm5iZiI6MTcyMDI4MjQ2OCwicGF0aCI6Ii83Njc2NDg0MS8zNDYyNjMzMjEtOWQ1YzIxMDYtMzBjYi00YWViLTg5YjItOTk5ZWYyOTI2YWUxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzA2VDE2MTQyOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTljMzUxNmY4NDhhNjM0YzFkNDgxODJhZGMxOTU4NmVhMjJhZDkwMGUwNzJlYzhiZTllMDYyMWY5ZTNlYTc3YmEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.tWV6XvLjTjkVJmgW5Cq2MqPRaVRmKb36OjezPJZm_NU)

this issue has been solved , but the latest release doesn't include the fix . you need to move to the git version

**Comment by ShadowS0ng** [comment]
> can confirm. Doesn't start at all, even with floating window workaround. Running under hyprland 0.41 on Arch
> 
> git version works tho

this issue has been solved , but the latest release doesn't include the fix . you need to move to the git version

**Comment by Yappaholic** [comment]
The problem still persists on NixOS, so if anyone uses it, you have to wait for the nixpkgs update, or search for the git flakes :(

**Comment by alembiq** [comment]
> The problem still persists on NixOS, so if anyone uses it, you have to wait for the nixpkgs update, or search for the git flakes :(

the flake is here: github:wez/wezterm?dir=nix

**Comment by Yappaholic** [comment]
> the flake is here: github:wez/wezterm?dir=nix

I'm trying to build it for the home-manager, but the error is still there, maybe you could give some advice?
``` nix
  inputs = {
    # Specify the source of Home Manager and Nixpkgs.
    nixpkgs.url = "github:nixos/nixpkgs/nixos-unstable";
    home-manager = {
      url = "github:nix-community/home-manager";
      inputs.nixpkgs.follows = "nixpkgs";
    };
    ags.url = "github:Aylur/ags";
    wezterm.url = "github:wez/wezterm?dir=nix";
  };

  outputs = { nixpkgs, home-manager, ... }@inputs:
    let
      system = "x86_64-linux";
      pkgs = nixpkgs.legacyPackages.${system};
    in {
      homeConfigurations."savvy" = home-manager.lib.homeManagerConfiguration {
        pkgs = import nixpkgs {inherit system; };

        #Ags
        extraSpecialArgs = {inherit inputs; };
        modules = [ ./home.nix ];
      };
    };

```

**Comment by Yappaholic** [comment]
Nevermind, cudos to this comment in wezterm repo:

https://github.com/wez/wezterm/pull/3547#issuecomment-1915820504

**Comment by alembiq** [comment]
> Nevermind, cudos to this comment in wezterm repo:
> 
> [#3547 (comment)](https://github.com/wez/wezterm/pull/3547#issuecomment-1915820504)

That post is better than anything I would be able to provide anyway :)
